nixos/modules/services/networking/postgrest.nix

```diff
@@ -0,0 +1,103 @@
+{ lib, pkgs, config, ... }:
+let
+  cfg = config.services.postgrest;
+  inherit (lib) mkOption types literalExample;
+in
+{
+  options.services.postgrest = {
+    enable = lib.mkEnableOption "postgrest";
+    postgresUser = mkOption {
+      type = types.string;
+      example = "root";
+      description = ''
+        Username for postgres admin.
+      '';  
+    };
+    postgresPassword = mkOption {
+      type = types.string;
+      default = "";
+      example = "password";
+      description = ''
+        Password for postgres admin.
+      '';
+    };
```

Comment:
Can I supply a passwordFile somehow?

Reply:
What I use currently is `db-uri = "@/secrets/db-uri"` in the postgrest config, which makes PostgREST read the whole db URI from file; I don't think there's support right now for reading just the password from file though.

Reply:
Would it suffice to leave this as is and put the onus on the user of the module to securely pass the credentials however they prefer?

Reply:
Currently it always generates a db-uri line from user and password, which makes it impossible to securely pass the credentials. So at the least it would need to be optional for the db-uri line to be generated.

I'd probably suggest providing only a plain db-uri option instead of user+password+host+port+database, but I'm sure there's other ways to approach this (e.g., have an extra db-uri option that overrides the generated form).

That said, I have no particularly idea of what a nixos module config *should* look like, and will defer to @SuperSandro2000's thoughts.

Reply:
> What I use currently is `db-uri = "@/secrets/db-uri"` in the postgrest config, which makes PostgREST read the whole db URI from file; I don't think there's support right now for reading just the password from file though.

If thats easier to do than thats also fine to do.